packages/next/src/lib/turbopack-warning.ts

```diff
@@ -158,13 +156,24 @@ export async function validateTurboNextConfig({
     Log.error('Unexpected error occurred while checking config', e)
   }
 
-  if (hasWebpackConfig && !hasTurboConfig) {
-    Log.warn(
-      `Webpack is configured while Turbopack is not, which may cause problems.`
-    )
-    Log.warn(
-      `See instructions if you need to configure Turbopack:\n  https://nextjs.org/docs/app/api-reference/next-config-js/turbopack\n`
+  // If the build was defaulted to Turbopack, we want to warn about possibly ignored webpack
+  // configuration. Otherwise the user explicitly picked turbopack and thus we expect that
+  // they have configured it correctly.
+  if (process.env.TURBOPACK === 'auto' && hasWebpackConfig && !hasTurboConfig) {
+    const configFile = rawNextConfig.configFileName ?? 'your Next config file'
+    Log.error(
+      `ERROR: This build is using Turbopack, with a \`webpack\` config and no \`turbopack\` config. This may be a mistake.
+
+  As of Next.js 16 turbopack is enabled by default and custom webpack configurations may need to be migrated to Turbopack.
+
+  NOTE: your \`webpack\` config may have been added by a configuration plugin.
+
+  To configure Turbopack, see https://nextjs.org/docs/app/api-reference/next-config-js/turbopack
+
+  TIP: Many applications work fine under Turbopack with no configuration, if that is the case for you, you can silence this error by passing the \`--turbopack\` or \`--webpack\` flag explicitly or simply setting an empty turbopack config in ${configFile} (e.g. \`turbopack: {}\`).`
```

Comment:
> or simply setting an empty turbopack config in ${configFile} (e.g. \`turbopack: {}\`).

empty object for turbopack config doesn't actually allow to skip this warning/error

`hasTurboConfig` is false when attemped because there is config flattening, and because empty turbopack config doesn't have any keys - flattened config will not actually have any turbopack properties

Reply:
[yeah merged a fix for this this morning. ](https://github.com/vercel/next.js/pull/84494) fixed that issue (with tests!)